frontend/src/store/modules/visitors/actions.js

```diff
@@ -34,13 +39,39 @@ export default {
         context.commit(SET_ACTIVE_BUTTON, button);
     },
     [CHANGE_FETCHED_BUTTON_STATE]: (context, data) => {
-
         if (data.value) {
             context.commit(SET_BUTTON_FETCHING, data.button);
         } else {
             context.commit(RESET_BUTTON_FETCHING, data.button);
         }
     },
+    [FETCH_BUTTON_DATA]: (context, data) => {
+        if (!data.value) {
+            return;
+        }
+        context.commit(SET_BUTTON_FETCHING, data.button);
+
+        const period = getTimeByPeriod(context.state.selectedPeriod);
+
+        return factoryVisitorsService.create(data.button)
+            .fetchButtonValue(period.startDate, period.endDate)
+            .then(response => {
+                context.commit(SET_BUTTON_DATA, data.button, response.data);
+                context.commit(RESET_BUTTON_FETCHING, data.button);
+            })
+            .catch(err => {
+                context.commit(RESET_BUTTON_FETCHING, data.button);
+                throw err;
+            });
+
+    },
+    [GET_EACH_BUTTON_DATA]: (context) => {
```

Comment:
I'm not sure that you need this, you can simply get everything from state or with mapState or with getter